extensions/api/query/src/main/java/org/eclipse/edc/catalog/api/query/FederatedCatalogCacheQueryApiExtension.java

```diff
@@ -0,0 +1,45 @@
+package org.eclipse.edc.catalog.api.query;
+
+import org.eclipse.edc.catalog.spi.QueryEngine;
+import org.eclipse.edc.runtime.metamodel.annotation.Extension;
+import org.eclipse.edc.runtime.metamodel.annotation.Inject;
+import org.eclipse.edc.spi.system.ServiceExtension;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+import org.eclipse.edc.spi.system.health.HealthCheckResult;
+import org.eclipse.edc.spi.system.health.HealthCheckService;
+import org.eclipse.edc.web.spi.WebService;
+
+@Extension(value = FederatedCatalogCacheQueryApiExtension.NAME)
+public class FederatedCatalogCacheQueryApiExtension implements ServiceExtension {
+
+    public static final String NAME = "Cache Query API Extension";
+
+    @Inject
+    private WebService webService;
+
+    @Inject
+    private QueryEngine queryEngine;
+
+    @Inject(required = false)
+    private HealthCheckService healthCheckService;
+
+    @Override
+    public String name() {
+        return NAME;
+    }
+
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        var catalogController = new FederatedCatalogApiController(queryEngine);
+        webService.registerResource(catalogController);
```

Comment:
it probably should, but then we're forced to provide port mappings for the management context, i.e. config values for `web.http.management.path` and `web.http.management.port`. 

That may not be intuitive, e.g. when running the FC in true standalone mode. 

How about instead of hard-coding the `"managment"` context we provide a config value, into which context the api controller should be registered?

Reply:
The management context has its own default values (defined in `ManagementApiConfigurationExtension`: port 8181 and path `/api/v1/management`, so by default there should be no need to configure them

Reply:
ok, in [this commit](https://github.com/eclipse-dataspaceconnector/FederatedCatalog/pull/13/commits/a988480ca62964be10b9bab48df410a4be73dffb) I switched it over to use the `ManamentApiConfiguration`. Is that what you meant?